### PR TITLE
disk_check: Script updated to run good in 201811 & 201911

### PR DESCRIPTION
**Pull request by renukamanavalan**
#### What I did
Have independent subdirs for each mounted dir to avoid any collisions of files/dirs by same name.
Adopt for older version of python3

#### How I did it
Changes:
1) Individual subdirs for each dir to be mounted
2) subprocess args made compatible with older version of python3 (tested in version 3.5.3)

#### How to verify it
1) Simulate read-only state
2) Run this script
3) Test ssh via new tacacs user (who had not logged in earlier)

### CODE REVIEW

**Comment by lguohan** [comment]
@qiluo-msft , can you take priority to review this one?

File: scripts/disk_check.py [hunk 1]
```diff
@@ -64,17 +69,17 @@ def test_writable(dirs):
...
+    proc = subprocess.run(cmd, shell=True, stdout=subprocess.PIPE)
     ret = proc.returncode
     if ret:
         log_err("failed: ret={} cmd={}".format(ret, cmd))
     else:
         log_info("ret={} cmd: {}".format(ret, cmd))
 
     if proc.stdout:
-        log_info("stdout: {}".format(str(proc.stdout)))
+        log_info("stdout: {}".format(proc.stdout.decode("utf-8")))
```
Comment (qiluo-msft) [inline scripts/disk_check.py:80]: >"utf-8" [](http://example.com/codeflow?start=56&length=7)

You can use default parameter value.

File: tests/disk_check_test.py [hunk 2]
```diff
@@ -26,7 +27,7 @@
         "workdir": "/tmp/tmpy",
         "mounts": "overlay_tmpx blahblah",
         "err": "/tmpx is not read-write|READ-ONLY: Mounted ['/tmpx'] to make Read-Write",
-        "cmds": ['mount -t overlay overlay_tmpx -o lowerdir=/tmpx,upperdir=/tmp/tmpx,workdir=/tmp/tmpy /tmpx']
+        "cmds": ['mount -t overlay overlay_tmpx -o lowerdir=/tmpx,upperdir=/tmp/tmpx/tmpx,workdir=/tmp/tmpy/tmpx /tmpx']
```
Comment (qiluo-msft) [inline tests/disk_check_test.py:30]: >upperdir=/tmp/tmpx/tmpx [](http://example.com/codeflow?start=66&length=23)

Could you explain why you need test code change when you change implementation details? Could you make test code not depends on the implementation?

File: scripts/disk_check.py (hunk 1 above)
Comment (renukamanavalan) [reply scripts/disk_check.py:80]: Prefer to keep it explicit. Any issue with that.

File: tests/disk_check_test.py (hunk 2 above)
Comment (renukamanavalan) [reply tests/disk_check_test.py:30]: This is a base change. We use explicit subdir under upper & work for each dir to mount.

**Review by qiluo-msft — APPROVED** [review]
Could you refine the PR title (and the title during squash merge)? It is not clear what you plan to adopt? Audience know nothing about 201811/201911 adoption requirement. I guess you mean adapt code to be compatible with python2?

**Pull request merged**